functions/channelCapture/lexClient.private.ts

```diff
@@ -101,8 +343,102 @@ export const deleteSession = async (
   }
 };
 
+export const deleteSession = async (
+  credentials: AWSCredentials,
+  {
+    botLanguage,
+    botSuffix,
+    enableLexV2,
+    environment,
+    helplineCode,
+    userId,
+  }: {
+    enableLexV2: boolean;
+    environment: string;
+    helplineCode: string;
+    botLanguage: string;
+    botSuffix: string;
+    userId: string;
+  },
+) => {
+  try {
+    if (enableLexV2) {
+      const result = await LexV2.getBotName(credentials, {
+        botLanguage,
+        botSuffix,
+        environment,
+        helplineCode,
+      });
+
+      if (result.status === 'failure') {
+        return result;
+      }
+
+      const { botAliasId, botId, localeId } = result.botDetails;
+
+      return await LexV2.deleteSession(credentials, {
+        botAliasId,
+        botId,
+        localeId,
+        sessionId: userId,
+      });
+    }
+
+    const botName = LexV1.getBotName({ botLanguage, botSuffix, environment, helplineCode });
+    const botAlias = 'latest'; // Assume we always use the latest published version
+
+    return await LexV1.deleteSession(credentials, { botAlias, botName, userId });
+  } catch (error) {
+    return {
+      status: 'failure',
+      error: error instanceof Error ? error : new Error(String(error)),
+    } as const;
+  }
+};
+
+export const isEndOfDialog = ({
+  enableLexV2,
+  lexResponse,
+}:
+  | {
+      enableLexV2: false;
+      lexResponse: LexRuntime.PostTextResponse;
+    }
+  | {
+      enableLexV2: true;
+      lexResponse: LexRuntimeV2.RecognizeTextResponse;
+    }) => {
+  if (enableLexV2) {
+    return LexV2.isEndOfDialog(lexResponse.sessionState?.dialogAction?.type);
+  }
+
+  return LexV1.isEndOfDialog(lexResponse.dialogState);
+};
+
+export const getBotMemory = ({
+  enableLexV2,
+  lexResponse,
+}:
+  | {
+      enableLexV2: false;
+      lexResponse: LexRuntime.PostTextResponse;
+    }
+  | {
+      enableLexV2: true;
+      lexResponse: LexRuntimeV2.RecognizeTextResponse;
+    }) => {
+  if (enableLexV2) {
+    return LexV2.convertV2ToV1Memory(lexResponse.sessionState?.intent?.slots);
+  }
+
+  return lexResponse.slots || {};
+};
+
 export type LexClient = {
```

Comment:
It might've been cleaner to have 2 versions of LexClient that implement a common type and use dependency injection to create the one you want to use, rather than the wrapper approach

But it's only temporary so I'm not asking for any changes, this is good